nsxt/resource_nsxt_policy_lb_http_monitor_profile.go

```diff
@@ -0,0 +1,184 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+func resourceNsxtPolicyLBHttpMonitorProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyLBHttpMonitorProfileCreate,
+		Read:   resourceNsxtPolicyLBHttpMonitorProfileRead,
+		Update: resourceNsxtPolicyLBHttpMonitorProfileUpdate,
+		Delete: resourceNsxtPolicyLBHttpMonitorProfileDelete,
+		Importer: &schema.ResourceImporter{
+			State: nsxtPolicyPathResourceImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":                getNsxIDSchema(),
+			"path":                  getPathSchema(),
+			"display_name":          getDisplayNameSchema(),
+			"description":           getDescriptionSchema(),
+			"revision":              getRevisionSchema(),
+			"tag":                   getTagsSchema(),
+			"request_body":          getLbMonitorRequestBodySchema(),
+			"request_header":        getLbHTTPHeaderSchema("Array of HTTP request headers"),
+			"request_method":        getLbMonitorRequestMethodSchema(),
+			"request_url":           getLbMonitorRequestURLSchema(),
+			"request_version":       getLbMonitorRequestVersionSchema(),
+			"response_body":         getLbMonitorResponseBodySchema(),
+			"response_status_codes": getLbMonitorResponseStatusCodesSchema(),
+			"fall_count":            getLbMonitorFallCountSchema(),
+			"interval":              getLbMonitorIntervalSchema(),
+			"rise_count":            getLbMonitorRiseCountSchema(),
+			"timeout":               getLbMonitorTimeoutSchema(),
+			"monitor_port":          getPolicyLbMonitorPortSchema(),
+		},
+	}
+}
+
+func resourceNsxtPolicyLBHttpMonitorProfileExists(id string, connector client.Connector, isGlobalManager bool) (bool, error) {
```

Comment:
Can we make this function common for all profile types as well?

Reply:
I think it's already calling the function resourceNsxtPolicyLBMonitorProfileExists which is common for all profile types?

Reply:
Yes, but rather than having 6 wrapper functions replicas for each monitor type, we can have just one - we can call it `resourceNsxtPolicyLBMonitorProfileExistsWrapper` or similar

Reply:
Done. Now we only use one wrapper function `resourceNsxtPolicyLBMonitorProfileExistsWrapper` across all monitor types